### ISSUE TITLE
Response examples doesn't work if status_code didn't passed

### ISSUE THREAD

**Issue by diego351** [tfranzel/drf-spectacular#875]
**Describe the bug**

In example like below status_code parameter can be inferred from responses dictionary
If status_codes didn't passed examples don't generate.

**To Reproduce**

```
  @extend_schema_view(
    get=extend_schema(
        responses={
            400: OpenApiResponse(
                response=ExceptionSerializer,
                description="My response description",
                examples=[
                    OpenApiExample(
                        name="DATE_PARSE_ERROR",
                        value={"api_status_code": "DATE_PARSE_ERROR", "extra": {"details": "asdf"}},
                       # status_codes=[400], <- if don't pass this the example won't work
                        description="My example 1 description",
                    ),
                ],
            ),
        },
    )
)
class ArchiveListView(ListAPIView):
....
```

**Expected behavior**
status_codes don't have to be duplicated. DRY improved

**Comment by Arcane97** [comment]
It took me a long time to figure out why OpenApiExample doesn't work, thank you dude.

**Comment by tfranzel** [comment]
Finally got around addressing this. examples in `OpenApiResponse` will now default to the correct values, and thus will show up as expected (if all filter parameters are just left on default)

Above example will now work without the extra `status_code` argument.

**Comment by tfranzel** [comment]
closed with release `0.26.0`
